BackEnd/routes.js

```diff
@@ -28,22 +29,32 @@ router.get('/config', (req, res, next) => {
 });
 
 router.post('/createGame', (req, res) => {
-    if (req.body.loginCode == null || req.body.orgName == null) {
+    if (req.body.loginCode == null || req.body.orgName == null ){//|| req.body.xCoord == null || req.body.yCoord == null || req.body.radius == null) {
 	res.sendStatus(400);
 	return;
     }
     Game.findOne({ gameCode: req.body.loginCode }, (err, game) => {
 	if (game)
 	    res.sendStatus(400);
 	else {
+		var newSafezone = new Safezone({location: [req.body.xCoord, req.body.yCoord],
+			radius: req.body.radius});
+		
 	    var newGame = new Game({ gameCode: req.body.loginCode,
 				     started: false,
-				     organizerName: req.body.orgName});
+				     organizerName: req.body.orgName,
+				     centralSafeZone: newSafezone._id});
+	    newSafezone.game = newGame._id;
 	    newGame.save((err) => {
 		if (err)
 		    res.sendStatus(500);
-		else
+		else {
+			newSafezone.save((err => {
+				if (err)
+					res.sendStatus(500)
+			}));
 		    res.sendStatus(200);
```

Comment:
you might want to move this line into the else clause of the newSafezone.save callback